src/Plugin/Field/ActivityEndDateFieldItemList.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Drupal\civicrm_entity\Plugin\Field;
+
+use Drupal\civicrm_entity\Entity\CivicrmEntity;
+use Drupal\Core\Field\FieldItemList;
+use Drupal\Core\TypedData\ComputedItemListTrait;
+use Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface;
+
+/**
+ * A computed field item list for Activities to provide an end date and time.
+ *
+ * In this class, you will notice timezone conversions from the default timezone
+ * to UTC. CiviCRM stores dates in the expected timezone, and Drupal always
+ * stores them in UTC. So we have to do a bit of conversion.
+ *
+ * This is also automatically done when a CivicrmEntity instance is saved and
+ * when an entity is loaded.
+ *
+ * @see \Drupal\civicrm_entity\Entity\CivicrmEntity::civicrmApiNormalize
+ * @see \Drupal\civicrm_entity\CiviEntityStorage::initFieldValues
+ */
+class ActivityEndDateFieldItemList extends FieldItemList {
+  use ComputedItemListTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function computeValue() {
+    $entity = $this->getEntity();
+    assert($entity instanceof CivicrmEntity);
+    $activity_date_time = $entity->get('activity_date_time')->value;
+    $duration = $entity->get('duration')->value;
+    if (!$activity_date_time) {
+      return;
+    }
+    // The time is already in UTC due to ::initFieldValues in storage.
+    // @see \Drupal\civicrm_entity\CiviEntityStorage::initFieldValues
+    $date = new \DateTime($activity_date_time, new \DateTimeZone('UTC'));
+    // We have to change this _back_ to the default timezone, as initFieldValues
+    // will be called again and it assumes the value is from CiviCRM, in the
+    // default timezone.
+    $date->setTimezone(new \DateTimeZone(date_default_timezone_get()));
```

Comment:
This is the workaround. When we computed the end date, we take the time as it is in UTC, convert it to the default timezone. We pass the time in this timezone to the field because `initFieldValues` will push it back to UTC for us.